scripts/build_docs.sh

```diff
@@ -7,9 +7,57 @@ set -euo pipefail
 # Build HTML documentation for the project
 # The output will be located in docs/docs
 
+# Function to validate and determine doc-gen4 revision
+determine_doc_gen_rev() {
+    local toolchain_content
+    local first_field
+    local second_field
```

Comment:
These variable names do not convey much information. How about something like:
```suggestion
    local toolchain_repository
    local toolchain_revision
```